src/apiCalls.js

```diff
@@ -0,0 +1,27 @@
+export const getProjects = async id => {
```

Comment:
This is great how you built it dynamically.  We may be able to work the palette GET calls into it as well, I can't decide if it'd become too bulky but its worth a shot.